projects/opentrain/gtfs_pipeline/heatmap/HeatMap.java

```diff
@@ -0,0 +1,895 @@
+                                 + (float) color1.getGreen());
+         int b =
+                 (int)
+                         (((float) color2.getBlue() - (float) color1.getBlue()) * value
+                                 + (float) color1.getBlue());
+
+         Color color = new Color(r, g, b, 50);
+         graphics2D.setPaint(color);
+         int x = 0;
+         int y = 0;
+         for (StopsAround stp : stopsAroundList) {
+           if (stp.i == i && stp.j == j) {
+             x =
+             getXpixel(stp.stop.getStopLat(), stp.stop.getStopLon());
+             y = getYpixel(stp.stop.getStopLat(), stp.stop.getStopLon());
+             //System.out.println("i= "+i+"j= "+j);
+             //if (x < image.getWidth() && y < image.getHeight()) {
+             // System.out.println("Painted"+"x="+x+"y="+y);
+             graphics2D.setPaint(Color.black);
+             graphics2D.setFont(new Font("TimesRoman", Font.BOLD, 28));
+             graphics2D.drawString("X", x, y);
+             // }
+             break;
+           }
+         }
+         graphics2D.setPaint(color);
+         x =
+         getXpixel((double) j / 100, (double) i / 100);
+         y = getYpixel((double) j / 100, (double) i / 100);
+         //if(x < image.getWidth() && y < image.getHeight())
+         graphics2D.fillRect(x, y, 5, 5);
+
+       }
+     }
+     graphics2D.dispose();
+     // Save file
+     boolean writeOnImage = false;
+     try {
+       writeOnImage =
+               ImageIO.write(
+                       image,
+                       "png",
+                       new File(
+                               heatMapPath.get()));
+     } catch (IOException e) {
+       e.printStackTrace();
+     }
+     if (!writeOnImage) {
+       System.out.println("can't write on image");
+     }
+     System.out.println("saved");
+   }*/
+
+  /*----Old functions x,y to long,lat-----
+
+  //y = 5.347393957·10-7 x12 + 3.503290175·10-7 x1 x2 + 9.781328537·10-10 x22 - 5.586224394·10-4 x1
+  // - 1.614915474·10-3 x2 + 32.34565226
+  public double getLongitudeByXY ( int x, int y)
+  {
+    double longitude = 5.347393957 * Math.pow(10, -7) * Math.pow(x, 2) + 3.503290175 * Math.pow(10, -7) * x * y
+            + 9.781328537 * Math.pow(10, -10) * Math.pow(y, 2) - 5.586224394 * Math.pow(10, -4) * x
+            - 1.614915474 * Math.pow(10, -3) * y
+            + 32.34565226;
+    return longitude;
+
+  }*/
+  // y = -1.28461208·10-6 x12 - 1.39031381·10-6 x1 x2
+  // - 3.786608865·10-7 x22 + 3.287492839·10-3 x1 + 9.395008917·10-4 x2 + 33.77106813
+  /* public double getLatitudeByXY ( int x, int y)
+  {
+    double latitude = -1.28461208 * Math.pow(10, -6) * Math.pow(x, 2) - 1.39031381 * Math.pow(10, -6) * x * y
+            - 3.786608865 * Math.pow(10, -7) * Math.pow(y, 2) + 3.287492839 * Math.pow(10, -3) * x
+            + 9.395008917 * Math.pow(10, -4) * y + 33.77106813;
+    return latitude;
+  }*/
+  private void arrivalTimeToLocationForMapNewFunction() {
```

Comment:
I guess this is called NewFunction because the original is commented out.
Please rename this after removing the comments.